frontend/app-development/features/processEditor/handlers/OnProcessTaskAddHandler.test.ts

```diff
@@ -1,25 +1,81 @@
 import type { Policy } from 'app-shared/types/Policy';
 import type { OnProcessTaskEvent } from '@altinn/process-editor/types/OnProcessTask';
-import type { TaskEvent } from '@altinn/process-editor/utils/ProcessTaskManager';
 import { OnProcessTaskAddHandler } from './OnProcessTaskAddHandler';
+import { BpmnTypeEnum } from '@altinn/process-editor/enum/BpmnTypeEnum';
+import type { TaskEvent } from '@altinn/process-editor/types/TaskEvent';
+import type { BpmnTaskType } from '@altinn/process-editor/types/BpmnTaskType';
+
+const orgMock = 'testOrg';
+const appMock = 'testApp';
+const currentPolicyMock: Policy = {
+  requiredAuthenticationLevelOrg: '3',
+  requiredAuthenticationLevelEndUser: '3',
+  rules: [],
+};
+const addLayoutSetMock = jest.fn();
+const mutateApplicationPolicyMock = jest.fn();
+const addDataTypeToAppMetadataMock = jest.fn();
+
+const createOnProcessTaskHandler = () =>
+  new OnProcessTaskAddHandler(
+    orgMock,
+    appMock,
+    currentPolicyMock,
+    addLayoutSetMock,
+    mutateApplicationPolicyMock,
+    addDataTypeToAppMetadataMock,
+  );
+
+const createTaskEvent = (extensionConfig?: object): TaskEvent =>
+  ({
+    element: {
+      id: 'testId',
+      businessObject: {
+        id: 'testEventId',
+        $type: BpmnTypeEnum.Task,
+        extensionElements: extensionConfig ? { values: [extensionConfig] } : undefined,
+      },
+    },
+  }) as TaskEvent;
 
 describe('OnProcessTaskAddHandler', () => {
-  it('should add default payment policy to current policy when task type is payment', () => {
-    const org = 'testOrg';
-    const app = 'testApp';
-    const currentPolicy: Policy = {
-      requiredAuthenticationLevelOrg: '3',
-      requiredAuthenticationLevelEndUser: '3',
-      rules: [],
-    };
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
 
-    const mutateApplicationPolicy = jest.fn();
+  it('should add layoutSet when data-task is added', () => {
+    const onProcessTaskAddHandler = createOnProcessTaskHandler();
+
+    onProcessTaskAddHandler.handleOnProcessTaskAdd({
+      taskEvent: createTaskEvent(),
+      taskType: 'data',
+    });
+
+    expect(addLayoutSetMock).toHaveBeenCalledWith({
```

Comment:
Great that you check what it has been called with! 🚀 
What do you think about also checking the number of times it was called to make sure we do not call the function more than once? 😄